src/main/java/nextstep/subway/common/GeneralExceptionHandler.java

```diff
@@ -0,0 +1,35 @@
+package nextstep.subway.common;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.servlet.NoHandlerFoundException;
+
+import nextstep.subway.line.exception.DuplicateLineException;
+import nextstep.subway.line.exception.NoSuchLineException;
+
+@ControllerAdvice
+public class GeneralExceptionHandler {
+
+	private final Logger log = LoggerFactory.getLogger(getClass());
+
+	@ExceptionHandler(NoHandlerFoundException.class)
+	public ResponseEntity<?> handleNotFoundException(Exception e) {
+		log.warn("핸들링 되지 않은 예외입니다.", e);
+		return ResponseEntity.notFound().build();
+	}
+
+	@ExceptionHandler(DuplicateLineException.class)
+	public ResponseEntity<?> duplicateException(Exception e) {
+		log.debug("중복된 값이 입력되었습니다.", e);
+		return ResponseEntity.badRequest().build();
+	}
+
+	@ExceptionHandler(NoSuchLineException.class)
+	public ResponseEntity<?> noSuchResourceException(Exception e) {
```

Comment:
크게 중요한 부분은 아니지만 Body로 아무런 내용이 나가고 있지 않으니 와일드카드 보다는 아래와 같이 처리해야 혼란이 적을 것 같네요.

```suggestion
	public ResponseEntity<Void> handleNotFoundException(Exception e) {
		log.warn("핸들링 되지 않은 예외입니다.", e);
		return ResponseEntity.notFound().build();
	}

	@ExceptionHandler(DuplicateLineException.class)
	public ResponseEntity<Void> duplicateException(Exception e) {
		log.debug("중복된 값이 입력되었습니다.", e);
		return ResponseEntity.badRequest().build();
	}

	@ExceptionHandler(NoSuchLineException.class)
	public ResponseEntity<Void> noSuchResourceException(Exception e) {
```